### PR TITLE
Add OWASP-functionality

### PR DESCRIPTION
**Pull request by bridgetlane**

### CODE REVIEW

**Comment by bridgetlane** [comment]
Added 2/4 of the rules endpoints, the other two are blocked by https://github.com/sethvargo/go-fastly/issues/54 or until I find an ok solution

**Comment by bridgetlane** [comment]
Added rule sets endpoints as well

**Comment by udryan10** [comment]
+1 
@sethvargo, anything holding this up from getting merged?

File: fastly/waf.go [hunk 1]
```diff
@@ -15,11 +14,11 @@ type WAFConfigurationSet struct {
...
-	ID                string     `jsonapi:"primary,waf"`
-	Version           int        `jsonapi:"attr,version"`
-	PrefetchCondition string     `jsonapi:"attr,prefetch_condition"`
-	Response          string     `jsonapi:"attr,response"`
-	LastPush          *time.Time `jsonapi:"attr,last_push"`
+	ID                string `jsonapi:"primary,waf"`
+	Version           int    `jsonapi:"attr,version"`
+	PrefetchCondition string `jsonapi:"attr,prefetch_condition"`
+	Response          string `jsonapi:"attr,response"`
+	LastPush          string `jsonapi:"attr,last_push"`
```
Comment (sethvargo) [inline fastly/waf.go]: Is there a reason this is a string instead of a time? It appears to be a time value.

Comment (bridgetlane) [reply fastly/waf.go]: @sethvargo  Yeah, because of issue #54

Comment (bridgetlane) [reply fastly/waf.go]: I could not get this to properly unmarshal into a time value, so I changed it to a string for the purposes of being able to use this while I was making changes. I'm open to feedback here also

**Pull request merged**